src/components/cuda_pw/common_defs.c

```diff
@@ -0,0 +1,106 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "common_defs.h"
+#include "debug_comp.h"
+#include "htable.h"
+
+#define ADDED_EVENTS_INITIAL_CAPACITY 64
+
+int initialize_dynamic_event_list_size(event_list_t * evt_table, int size)
+{
+    evt_table->capacity = size;
+    evt_table->count = 0;
+    evt_table->evts = (struct eventname_id_s *) calloc (evt_table->capacity, sizeof(struct eventname_id_s));
+    if (evt_table->evts == NULL) {
+        ERRDBG("Error allocating memory for dynamic event table.\n");
+        return PAPI_ENOMEM;
+    }
+    htable_init(&(evt_table->htable));
+    return PAPI_OK;
+}
+
+int initialize_dynamic_event_list(event_list_t * evt_table)
+{
+    return initialize_dynamic_event_list_size(evt_table, ADDED_EVENTS_INITIAL_CAPACITY);
+}
+
+static int reallocate_array(event_list_t *evt_table)
+{
+    unsigned int new_capacity = evt_table->capacity * 2;
+    struct eventname_id_s *tmp = (struct eventname_id_s *) calloc(new_capacity, sizeof(struct eventname_id_s));
```

Comment:
what is wrong with `realloc`? Also, use the PAPI version of allocation functions (i.e. `papi_realloc`).